src/components/shared/CallNodeContextMenu.js

```diff
@@ -355,6 +360,19 @@ class CallNodeContextMenu extends PureComponent<Props, State> {
     }
   }
 
+  _mouseDownHandler(event: MouseEvent): void {
+    event.preventDefault();
+  }
+
+  componentDidUpdate() {
+    if (this.state.isShown && this._contextMenu && this._contextMenu.menu) {
+      this._contextMenu.menu.addEventListener(
+        'mousedown',
+        this._mouseDownHandler
+      );
+    }
+  }
```

Comment:
My main problem with this approach is that we rely on some internal property of `_contextMenu`, this is very fragile.
Also we never remove this listener, but maybe that's OK.

But maybe you can call `event.preventDefault` in `handleClick`? Have you tried this?
Also I see that `onHide` also gets an event as parameter, but I don't think this one will work.

Reply:
So I tried adding `preventDefault` in `handleClick`. I think it's not working because the focusing is being caused by the mousedown event and not the click event. [Made a fiddle to show this.](https://jsfiddle.net/drabbitzz/k9jp2hm5/5/) Unfortunately react-contextmenu doesn't expose a prop for passing a function that will be executed after a mousedown event, similar to how it does for `onClick`.

Using `ReactDOM.findDOMNode(this._contextMenu)` instead of `this._contextMenu.menu` works as an alternative. I know its advised against but it doesn't look like its possible to do using refs without being able to modify the child component's code ([discussed here](https://reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components)). Other than this I'm a bit stumped for a different way to solve this.

As for the event listeners I was thinking about removing them in `componentWillUnmount` but the component seems to stay mounted the whole time. They could be removed in `compomentDidUpdate` when `isShown` is false if you think thats okay.

Reply:
Ah I see, thanks for the clear explanation!

Using `ReactDOM.findDOMNode` seems less brittle: at least we don't depend on some internal property. Of course this isn't perfect that that's how it is. Maybe we should also file a bug at the library explaining our problems... who knows, they might fix it directly there?

I think it's ok to not remove the listener, but please add a comment explaining why it's ok :)

Reply:
For now I just used `findDOMNode` and added some comments to explain. Also [opened an issue regarding the context menu](https://github.com/vkbansal/react-contextmenu/issues/263). Not sure if you would rather wait for that or not. I offered to make the PR so maybe it can get resolved quickly.